diesel/src/macros/mod.rs

```diff
@@ -796,12 +791,7 @@ macro_rules! __diesel_table_impl {
             /// Contains all of the columns of this table
             pub mod columns {
                 use super::table;
-                use $crate::{Expression, SelectableExpression, AppearsOnTable, QuerySource};
-                use $crate::backend::Backend;
-                use $crate::query_builder::{QueryFragment, AstPass, SelectStatement};
-                use $crate::query_source::joins::{Join, JoinOn, Inner, LeftOuter};
-                use $crate::query_source::{AppearsInFromClause, Once, Never, Pick};
-                use $crate::result::QueryResult;
+                use $crate::QuerySource;
```

Comment:
Is there a specific reason why we need to import `QuerySource` here?

Reply:
That's why `from_clause` defined in `QuerySource` is used in the following lines. However, since we don't need to import `QuerySource` globally, I've imported it in functions that need it.

https://github.com/diesel-rs/diesel/blob/d7ce43d2130bca434cd1acc2943a0c944d1deba4/diesel/src/macros/mod.rs#L38

https://github.com/diesel-rs/diesel/blob/d7ce43d2130bca434cd1acc2943a0c944d1deba4/diesel/src/macros/mod.rs#L831